remote-processor/RemoteCommandHandlerTemplate.h

```diff
@@ -112,144 +136,107 @@ class TRemoteCommandHandlerTemplate : public IRemoteCommandHandler
-
-            return false;
-        }
-
-        if (remoteCommand.getCommand() == "help") {
-
-            helpCommandProcess(strResult);
+    /** Handle a remote command
+     *
+     * @param[in] remoteCommand the remote command to handle
+     * @param[out] result string containing the result of the handled command
+     * @result true on success, false otherwise
+     */
+    bool remoteCommandProcess(const IRemoteCommand& remoteCommand, std::string& result)
+    {
+        try {
+            // Dispatch
+            const RemoteCommandParserItem& remoteCommandParserItem =
+                _remoteCommandParserItems.at(remoteCommand.getCommand());
 
-            return true;
+            return remoteCommandParserItem.parse(_commandParser, remoteCommand, result);
         }
+        catch (const std::out_of_range&) {
 
-        return pRemoteCommandParserItem->parse(_pCommandParser, remoteCommand, strResult);
-    }
-
-    // Max command usage length, use for formatting
-    void initMaxCommandUsageLength()
-    {
-        if (!_uiMaxCommandUsageLength) {
-            // Show usages
-            uint32_t uiIndex;
-
-            for (uiIndex = 0; uiIndex < _remoteCommandParserVector.size(); uiIndex++) {
+            if (remoteCommand.getCommand() == helpCommand) {
 
-                const CRemoteCommandParserItem* pRemoteCommandParserItem = _remoteCommandParserVector[uiIndex];
+                help(result);
 
-                uint32_t uiRemoteCommandUsageLength = pRemoteCommandParserItem->usage().length();
+                return true;
+            }
 
-                if (uiRemoteCommandUsageLength > _uiMaxCommandUsageLength) {
+            // Not found
+            result = "Command not found!\nUse \"help\" to show available commands";
 
-                    _uiMaxCommandUsageLength = uiRemoteCommandUsageLength;
-                }
-            }
+            return false;
         }
     }
 
-    /////////////////// Remote command parsers
-    /// Help
-    void helpCommandProcess(std::string& strResult)
+    /** Format help display
+     *
+     * @param result the formatted help string
+     */
+    void help(std::string& result)
     {
-        initMaxCommandUsageLength();
-
-        strResult = "\n";
-
-        // Show usages
-        uint32_t uiIndex;
-
-        for (uiIndex = 0; uiIndex < _remoteCommandParserVector.size(); uiIndex++) {
+        std::vector<std::pair<std::string, std::string> > helps;
+        size_t maxUsage = 0;
 
-            const CRemoteCommandParserItem* pRemoteCommandParserItem = _remoteCommandParserVector[uiIndex];
-
-            std::string strUsage = pRemoteCommandParserItem->usage();
-
-            strResult += strUsage;
-
-            // Align
-            uint32_t uiToSpacesAdd = _uiMaxCommandUsageLength + 5 - strUsage.length();
-
-            while (uiToSpacesAdd--) {
+        for (auto item : _remoteCommandParserItems) {
```

Comment:
auto &

 in c++11, `item` is taken by copy, put `auto &` to take by reference.